lib/puma/server.rb

```diff
@@ -346,11 +348,12 @@ def handle_servers
                 rescue IO::WaitReadable
                   next
                 end
-                drain += 1 if shutting_down?
-                pool << Client.new(io, @binder.env(sock)).tap { |c|
-                  c.listener = sock
-                  c.send(addr_send_name, addr_value) if addr_value
-                }
+                drain += 1 if shd
```

Comment:
Actually you are right, it can possibly affect the total connections count written to Puma log (line 367 `@log_writer.debug "Drained #{drain} additional connections." if drain`).
Do you want me to revert that change?

Reply:
Yes, I would prefer correct before performance but maybe there are other opinions

Reply:
> Yes, I would prefer correct before performance but maybe there are other opinions

I reverted my changes :)